src/NexusMods.Collections/CollectionCreator.cs

```diff
@@ -59,6 +76,92 @@ private static string GenerateNewCollectionName(string[] allNames)
         return result.Remap(group);
     }
 
+    public static async ValueTask<CollectionRevisionMetadata.ReadOnly> UploadCollectionRevision(IServiceProvider serviceProvider, LoadoutItemGroupId groupId, CancellationToken cancellationToken)
+    {
+        var connection = serviceProvider.GetRequiredService<IConnection>();
+        var loginManager = serviceProvider.GetRequiredService<ILoginManager>();
+        var libraryService = serviceProvider.GetRequiredService<ILibraryService>();
+        var nexusModsLibrary = serviceProvider.GetRequiredService<NexusModsLibrary>();
+        var temporaryFileManager = serviceProvider.GetRequiredService<TemporaryFileManager>();
+        var jsonSerializerOptions = serviceProvider.GetRequiredService<JsonSerializerOptions>();
+        var mappingCache = serviceProvider.GetRequiredService<IGameDomainToGameIdMappingCache>();
```

Comment:
Should the creator just be a singleton and these services be readonly state? I think this would make sense if these services needed to be transient, but it seems to me they are all singletons.